src/JWT/Algorithms/RSAlgorithmFactory.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace JWT.Algorithms
+{
+    public sealed class RSAlgorithmFactory : HMACSHAAlgorithmFactory
+    {
+        private readonly Func<X509Certificate2> _certFactory;
+
+        public RSAlgorithmFactory(Func<X509Certificate2> certFactory)
```

Comment:
Totally agree, there is no need to make it more complex, even though there are many RS signatures. Maybe i submit the pull request for the other ones ;). I'm getting hyped.